pkg/transport/generic_transport_controller.go

```diff
@@ -62,7 +62,13 @@ func NewTransportController(ctx context.Context, bindingInformer controlv1alpha1
 	if err != nil {
 		return nil, fmt.Errorf("failed to get wrapped object GVR - %w", err)
 	}
+	return NewTransportControllerForGVR(ctx, bindingInformer, transport, wdsClientset, wdsDynamicClient, transportClientset, transportDynamicClient, wdsName, wrappedObjectGVR), nil
+}
 
+// NewTransportController returns a new transport controller
+func NewTransportControllerForGVR(ctx context.Context, bindingInformer controlv1alpha1informers.BindingInformer, transport Transport,
```

Comment:
can we put this function private? (start with lower case)
unit test file should still be able to call it cause it is the same package right?

in addition, the name `ForGVR` is not so clear (GVR of what?). 
can we call it `ForWrappedObjectGVR`?

Reply:
I changed the variable name and put a real explanation in the comment.

Reply:
I changed the func name as suggested.
I think this func is something that is reasonable for non-test clients to use.
I added some explanation in the func comments.

Reply:
The second func is also simpler in that it can not return an error.

Reply:
A caller of the second func is less complex --- no need to construct the k8s clientset, no need to handle an error.